### ISSUE TITLE
downloaded modules missing from cache

### ISSUE THREAD

**Issue by tvald** [gruntwork-io/terragrunt#512]
**Bug Description**
I'm seeing an intermittent issue where previously-downloaded modules are missing from the cache in the temporary directory that Terragrunt creates. The issue arises only after a period where I don't run `terragrunt` for a while, at least more than 48 hours (not sure exactly). The error looks like this:

```
[terragrunt] 2018/07/05 19:00:15 Setting working directory to C:/Windows/Temp/terragrunt/pT9Mpllsp6HPkGyxi7aF0jz7mAY/KIjyWX_Q_zf4EaXmi8KIqCY0oJ0
<...>
Initializing modules...
- module.secure_policy
Error downloading modules: Error loading modules: module secure_policy: open .terraform\modules\fc4bc6f16751f32d1a52829498208460: The system cannot find the file specified.
[terragrunt] 2018/07/05 19:00:15 Hit multiple errors:
exit status 1
```

**System Configuration**
Windows 10, version 1803, build 17134.112 (latest?)

**Workaround**
Run terragrunt with `--terragrunt-source-update` to force re-download of module source.

**Related Issues**
#266, #301

**More Details**
Inspection of the temporary directory's cache shows it to be empty:
```
Directory: C:\Windows\Temp\terragrunt\pT9Mpllsp6HPkGyxi7aF0jz7mAY\KIjyWX_Q_zf4EaXmi8KIqCY0oJ0\.terraform\modules

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----         7/1/2018  10:11 PM            201 modules.json
```

Although `modules.json` contains a zombie entry for the missing module:
```
PS > type modules.json
{"Modules":[{"Source":"./config/s3-bucket-policy/secure","Key":"1.secure_policy;./config/s3-bucket-policy/secure","Version":"","Dir":".terraform\\modules\\fc4bc6f16751f32d1a52829498208460","Root":""}]}
```

My guess is that either Windows is clearing the module's subdirectory for some reason, or `terragrunt` is somehow unintentionally deleting it.

**Comment by brikis98** [comment]
How weird. We used to write the cache directory to a temp folder, but temp would get cleared by most operating systems on a regular basis, so we switched to writing to the home directory instead. I haven't seen this issue on *nix systems since then. 

Does Windows delete stuff in its home directory? I'm not aware of any reason Terragrunt would do it after a couple days... Perhaps paste the whole log output?

**Comment by tvald** [comment]
Thanks for looking at this.

> so we switched to writing to the home directory instead

Hmm, I don't see anything related to `terragrunt` or `terraform` in my home or documents directory. What is the expected path on Windows where downloaded source for modules is cached? (NB: my terminology may be off here, since I just started using `terragrunt`...)

I did configure a shared plugin cache, but that only contains provider plugins.

Here's the full log output (with anonymized paths):
```
[terragrunt] [C:\Users\username\Documents\git\ops-infrastructure\master\assets\live\__terraform__\remote_state] 2018/07/05 19:00:12 Running command: terraform --version
[terragrunt] 2018/07/05 19:00:12 Reading Terragrunt config file at C:/Users/username/Documents/git/ops-infrastructure/master/assets/live/__terraform__/remote_state/terraform.tfvars
[terragrunt] 2018/07/05 19:00:12 Cleaning up existing *.tf files in C:/Windows/Temp/terragrunt/pT9Mpllsp6HPkGyxi7aF0jz7mAY/KIjyWX_Q_zf4EaXmi8KIqCY0oJ0
[terragrunt] 2018/07/05 19:00:12 Downloading Terraform configurations from git::file://C:/Users/username/Documents/git/ops-infrastructure/master?ref=b7907ec94b905098df9a7bb811b73278348c817a into C:/Windows/Temp/terragrunt/pT9Mpllsp6HPkGyxi7aF0jz7mAY/KIjyWX_Q_zf4EaXmi8KIqCY0oJ0 using terraform init
[terragrunt] [C:\Users\username\Documents\git\ops-infrastructure\master\assets\live\__terraform__\remote_state] 2018/07/05 19:00:12 Initializing remote state for the s3 backend
[terragrunt] [C:\Users\username\Documents\git\ops-infrastructure\master\assets\live\__terraform__\remote_state] 2018/07/05 19:00:13 Running command: terraform init -backend-config=bucket=MyProject-terraform -backend-config=key=__terraform__/remote_state/terraform.tfstate -backend-config=region=us-east-1 -backend-config=encrypt=true -backend-config=dynamodb_table=tf-lock-table -from-module=git::file://C:/Users/username/Documents/git/ops-infrastructure/master?ref=b7907ec94b905098df9a7bb811b73278348c817a C:/Windows/Temp/terragrunt/pT9Mpllsp6HPkGyxi7aF0jz7mAY/KIjyWX_Q_zf4EaXmi8KIqCY0oJ0
Copying configuration from "git::file://C:/Users/username/Documents/git/ops-infrastructure/master?ref=b7907ec94b905098df9a7bb811b73278348c817a"...
Terraform initialized in an empty directory!

The directory has no Terraform configuration files. You may begin working
with Terraform immediately by creating Terraform configuration files.
[terragrunt] 2018/07/05 19:00:15 Copying files from C:/Users/username/Documents/git/ops-infrastructure/master/assets/live/__terraform__/remote_state into C:/Windows/Temp/terragrunt/pT9Mpllsp6HPkGyxi7aF0jz7mAY/KIjyWX_Q_zf4EaXmi8KIqCY0oJ0
[terragrunt] 2018/07/05 19:00:15 Setting working directory to C:/Windows/Temp/terragrunt/pT9Mpllsp6HPkGyxi7aF0jz7mAY/KIjyWX_Q_zf4EaXmi8KIqCY0oJ0
[terragrunt] [C:\Users\username\Documents\git\ops-infrastructure\master\assets\live\__terraform__\remote_state] 2018/07/05 19:00:15 Initializing remote state for the s3 backend
[terragrunt] [C:\Users\username\Documents\git\ops-infrastructure\master\assets\live\__terraform__\remote_state] 2018/07/05 19:00:15 Running command: terraform init -backend-config=encrypt=true -backend-config=dynamodb_table=tf-lock-table -backend-config=bucket=MyProject-terraform -backend-config=key=__terraform__/remote_state/terraform.tfstate -backend-config=region=us-east-1
Initializing modules...
- module.secure_policy
Error downloading modules: Error loading modules: module secure_policy: open .terraform\modules\fc4bc6f16751f32d1a52829498208460: The system cannot find the file specified.
[terragrunt] 2018/07/05 19:00:15 Hit multiple errors:
exit status 1
```

**Comment by brikis98** [comment]
What version of Terragrunt are you on?

**Comment by tvald** [comment]
> What version of Terragrunt are you on?

```
PS> terragrunt --version
terragrunt version v0.15.2
```

**Comment by brikis98** [comment]
Ah, I figured it out. The issue is here:

https://github.com/gruntwork-io/terragrunt/blob/1bbbbf2737579f6f80e7fb67f1d152c8f55ba209/options/options.go#L100-L107

We had updated Terragrunt to store downloads in the `$HOME` dir, but accidentally left an `if-statement` where, for Windows, we still put it in a temp folder! Would you be up for a quick PR to remove that if-statement and use the `$HOME` dir on all operating systems?

**Comment by tvald** [comment]
Sure, I'll take a look this weekend.

**Comment by tvald** [comment]
Fix pending in #515.
